lib/active_merchant/billing/gateways/borgun.rb

```diff
@@ -180,19 +180,30 @@ def headers
         }
       end
 
-      def build_request(action, post)
+      def build_request(action, post, options={})
         mode = action == 'void' ? 'cancel' : 'get'
         xml = Builder::XmlMarkup.new indent: 18
         xml.instruct!(:xml, version: '1.0', encoding: 'utf-8')
         xml.tag!("#{mode}Authorization") do
           post.each do |field, value|
             xml.tag!(field, value)
           end
+          build_airline_xml(xml, options[:passenger_itinerary_data]) if options[:passenger_itinerary_data]
         end
         inner = CGI.escapeHTML(xml.target!)
         envelope(mode).sub(/{{ :body }}/, inner)
       end
 
+      def build_airline_xml(xml, airline_data)
+        xml.tag!('PassengerItineraryData') do
```

Comment:
One little less "hinky" way to do this might be to just call `xml.PassengerItineraryData` (I think the `tag` is only needed if the element name is a reserved word). For that reason, might be worth keeping the `tag!` on line 201 (since that element name comes from user input).

Reply:
So the hinkiness I had in mind in particular was having a random extra helper buried 3 methods away from where it actually gets called (because of the iterative `xml.tag!` in `build_request` that doesn't allow for nesting). 

I'd be happy to swap out the `tag!`s but am a little 🤔 about having two different functions that serve the same purpose; what do you think?

Reply:
Ah, I see what you're saying. I think it's fine to have one method.